### PR TITLE
Restic backup

### PR DESCRIPTION
**Pull request by kayiwa**
related to https://github.com/pulibrary/princeton_ansible/issues/4193

### CODE REVIEW

File: services/restic_backup.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+     ### repository on google cloud
+     export GOOGLE_APPLICATION_CREDENTIALS='/var/lib/postgresql/.restic/pul-gcdc-filename.json'
+
+     export RESTIC_ARCHIVE_REPOSITORY='gs:postgres-version-backup:yourpath'
+
+     export RESTIC_REPOSITORY=$RESTIC_ARCHIVE_REPOSITORY
+     export RESTIC_PASSWORD_FILE='/var/lib/postgresql/.restic.pwd'
+     ```
+
+     The `/var/lib/postgresql` path above will be `/home/pulsys` for mariadb
```
Comment (acozine) [inline services/restic_backup.md:57]:
```suggestion
     If you are backing up postgreSQL, use the path `/var/lib/postgresql`.
     If you are backing up mariadb, use the path `/home/pulsys`.
```

File: services/restic_backup.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      ```
+     Add a line that points to the `full_pg_backup.sh` script
+
+    ```file
+    0 5 * * * /var/lib/postgresql/.restic/full_pg_backup.sh
+    ```
+    
+For mariadb do the following:
+
+  1. Copy all the files under the [mariadb](mariadb) to the `pulsys` user
```
Comment (acozine) [inline services/restic_backup.md:97]:
```suggestion
  1. Copy all the files under the [mariadb](mariadb) to the `pulsys` user:
```

File: services/restic_backup.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     ```
+  2. Make all the scripts executable by the `pulsys` user:
+     ```bash
+     chmod u+x ~/.restic/maria_backup.sh
+     ```
+  3. Set Up automated backups by creating a cron job for the `pulsys` user with the following:
+     ```bash
+      crontab -e
+      ```
+     Add a line that points to the `full_pg_backup.sh` script
```
Comment (acozine) [inline services/restic_backup.md:109]:
```suggestion
     Add a line that points to the `full_pg_backup.sh` script:
```

File: services/restic_backup.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+    ```file
+    0 5 * * * /home/pulsys/.restic/maria_backup.sh
+    ```
+
+## Restore a backup
```
Comment (acozine) [inline services/restic_backup.md:115]:
```suggestion
## Restore from a backup
```

**Review by acozine — COMMENTED** [review]
Generally looks good, and thanks for documenting this. I found a bunch of copyediting suggestions and one or two questions.

File: services/gce_bucket.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+You can find the complete documentation on [Creating a Bucket](https://cloud.google.com/storage/docs/creating-buckets). 
+
+Here's a cheatsheet for the usual steps we do:
+
+  1. In the Google Cloud Console, go to [Buckets](https://console.cloud.google.com/storage/browser?_ga=2.76486933.1698741265.1707489780-289047190.1707229976).
+  2. Click **+ Create.**
+  3. On the **Create a bucket** page, enter your bucket information. To go to the next step, click **Continue.**
+     * For **Name your bucket**, enter a name that meets the [bucket name requirements](https://cloud.google.com/storage/docs/buckets#naming).
+     *  For **Choose where to store your data**, select a [Location type](https://cloud.google.com/storage/docs/locations) and [Location](https://cloud.google.com/storage/docs/locations#available-locations) where the bucket data will be permanently stored.
+     *  For **Choose a storage class for your data**, either select a [default storage class](https://cloud.google.com/storage/docs/storage-classes) for the bucket.
```
Comment (acozine) [inline services/gce_bucket.md:12]:
```suggestion
     *  For **Choose a storage class for your data**, select Standard as the [default storage class](https://cloud.google.com/storage/docs/storage-classes) for the bucket.
```

**Comment by acozine** [comment]
Holding until someone has tested these instructions by backing up a database and restoring from a backup.

**Review by kayiwa — COMMENTED** [review]
thanks @christinach

**Review by acozine — APPROVED** [review]
Restore worked today, thanks @christinach!

**Pull request merged**